### DISCUSSION TITLE
Dependency `autocurator` is not maintained, only supports Linux and Python <=3.8

### DISCUSSION THREAD

**Discussion by chengzhuzhang** [ESGF/esg-publisher#219] [General]
The E3SM team is using esg-publisher for data publication to ESGF. However, we found it that there is conflict in maintaining esg-publisher dependencies with E3SM's own libraries in one conda environment. The current resolution is to maintain a separate environment for data publication. The main problem is that `autocurator` is not maintained, only supports Linux and Python <=3.8, details are documented by @tomvothecoder here. https://github.com/E3SM-Project/datasm/issues/143. I'm posting here to raise awareness and hopefully we can find a solution. Thank you!

**Comment by sashakames** [comment]
@chengzhuzhang @tomvothecoder  Have you upgraded to v5.2.0 yet?  This supports xarray.
See https://esg-publisher.readthedocs.io

**Reply by sashakames** [reply]
Additionally: (1) we dropped a conda requirement and just distribute via pip
(2) You will need to migrate your config format from .ini to .yaml but I'd hope you agree this is a step forward though a one-time inconvenience.

**Reply by tomvothecoder** [reply]
Thanks for the reply Sasha! We have not tried `esgcet=5.2.0` yet. In the [v5.2.0 release notes](https://esg-publisher.readthedocs.io/en/refactor/whatsnew.html#v5-2-0) I see that `autocurator` can be substituted with Xarray using the `-xarray` flag. I opened a [PR](https://github.com/E3SM-Project/datasm/pull/144) in the datasm repo for @TonyB9000 to start experimenting.

> Additionally: (1) we dropped a conda requirement and just distribute via pip

I am curious why pip is now preferred over conda for distributing `esgcet` and did you consider distribution on [conda-forge](https://conda-forge.org/#about)?

Just as an FYI, if a package (e.g. `my_package`) has a dependency that is only available PyPi/pip (e.g., `esgcet`), then `my_package` cannot be built and distributed on Anaconda/conda. This won't be a problem for the [datasm repo](https://github.com/E3SM-Project/datasm) in particular since it is not a conda package, but it can be for other conda packages.

Also a `conda install` command in the [TLDR doc](https://esg-publisher.readthedocs.io/en/v5.2.0/#tl-dr) includes `esgcet`, which will install an older version first. The subsequent `pip install esgcet` will update it to v5.2.0 though.

> (2) You will need to migrate your config format from .ini to .yaml but I'd hope you agree this is a step forward though a one-time inconvenience.

We can add a `pip` section to our conda-env yaml files ([example](https://github.com/E3SM-Project/datasm/pull/144/files#diff-8b7144213025eca9bb90e4ddaf0e9e2a5a7f8140305294383c0678c14da8638dR35-R36)).

**Reply by sashakames** [reply]
Debugging the conda-forge recipes is too time consuming for now until we get additional developer support from ORNL if we are to continue support for esgcet long-term.  We may migrate to a different means of publishing (Globus Flows) if they'll work at all sites.

I'll consider fixing the TLDR, but those steps should work even though installing and updating esgcet might be confusing.

**Reply by tomvothecoder** [reply]
Thank you Sasha, that makes sense to me. E3SM is fortunate to have a few people who can and is willing to manage conda-forge feedstocks.

No big deal on the documentation since it still works fine.

**Reply by sashakames** [reply]
If conda-forge would much preferred method for you and there's time on your/E3SM to create the feedstock, I'd be very happy to accept the PR for the esg-publisher side.

**Reply by tomvothecoder** [reply]
The pip solution works fine for the `datasm` repo right now. If we ever need `esgcet` on `conda-forge` and it is not available on there by then, it is a good opportunity for @TonyB9000 to try creating and maintaining a feedstock repo (he's the main user of `esgcet`).

**Reply by TonyB9000** [reply]
@tomvothecoder @sashakames Agreed!  I know that Tom has provided extensive instructions on this - but until one actually works through the process - it remains mysterious. 

("he's the main user of esgcet" - I had no idea!)
("I opened a PR in the datasm repo for @TonyB9000 to start experimenting." - ???)

Bad git-practice on my part - I process repo almost entirely from the acme1-console side, and only visit the git web stuff when I PR.

**Reply by tomvothecoder** [reply]
> ("he's the main user of esgcet" - I had no idea!)

I should have been more specific and mean from the E3SM Publication Team (you, me and Jill).

> ("I opened a PR in the datasm repo for @TonyB9000 to start experimenting." - ???)

I hyperlinked the [PR](https://github.com/E3SM-Project/datasm/pull/144) in that [original comment](https://github.com/ESGF/esg-publisher/discussions/219#discussioncomment-6723988).
